platforms/documentation/docs/src/docs/userguide/reference/dsl-apis/public_apis.adoc

```diff
@@ -18,192 +18,68 @@
 
 Gradle provides many public APIs that allow developers to interact with various build system components.
 
-Any class name containing `Internal`, or that is located in a package containing the segment `.internal.` is by definition not part of the public Gradle API and should not be relied upon for stable usage.
-
-The following table summarizes public Gradle APIs and their common use cases.
-This is *not* an exhaustive list:
-
-[cols="1,1,2"]
+Any class or package name containing `Internal`, or located in a package segment `.internal.`, is not part of the public Gradle API and should not be relied upon for stable usage.
+Use of Gradle internal APIs in plugins and build scripts can break builds when either Gradle or plugins change.
+
+The list of public API packages is maintained in the link:https://github.com/gradle/gradle/blob/master/buildSrc/subprojects/configuration/src/main/kotlin/org/gradle/gradlebuild/public-api.kt[Gradle public API definition] (excluding any subpackages containing internal).
```

Comment:
I don't know if we should document this here. It's an implementation detail of our build, and this link could easily break when moving it around. I think just including the list is fine.